common-utils/src/main/java/eu/arrowhead/common/http/filter/authorization/BlacklistFilter.java

```diff
@@ -1,40 +1,84 @@
 public class BlacklistFilter extends ArrowheadFilter {
+	
+	//=================================================================================================
+	// members
+	
+	@Autowired
+	protected SystemInfo sysInfo;
+	
+	@Autowired
+	protected ArrowheadHttpService arrowheadHttpService;
+	
+	@Value(Constants.$FORCE_BLACKLIST_FILTER_WD)
+	private boolean force;
 
 	//=================================================================================================
 	// assistant methods
 
 	//-------------------------------------------------------------------------------------------------
 	@Override
-	protected void doFilterInternal(final HttpServletRequest request, final HttpServletResponse response, final FilterChain chain) throws IOException, ServletException {
-		try {
-
-			Object systemName = request.getAttribute(Constants.HTTP_ATTR_ARROWHEAD_AUTHENTICATED_SYSTEM);
-
-			// TODO this filter should lookup for the blacklist service and use it against to the system name
-			// if system name is found on the blacklist throw ForbiddenException
-			// if requester is sysop, no need for check
-			// if request is for lookup for authentication, no need for check
-			// blacklist entry can be temporary (with expiration)
-
-			chain.doFilter(request, response);
-		} catch (final ArrowheadException ex) {
-			handleException(ex, response);
+	protected void doFilterInternal(final HttpServletRequest request, final HttpServletResponse response, final FilterChain chain) throws IOException, ServletException {	
+		log.debug("BlacklistFilter is active");
+		
+		// if requester is sysop, no need for check
+		final boolean isSysop = Boolean.valueOf(request.getAttribute(Constants.HTTP_ATTR_ARROWHEAD_SYSOP_REQUEST).toString());
+		
+		// if request is for lookup for authentication, no need for check
+		final boolean isAuthLookup = false; //TODO
+		
+		if (!isSysop && !isAuthLookup) {
+			log.debug("checking Blacklist");
+			try {
+					
+				final String systemName = request.getAttribute(Constants.HTTP_ATTR_ARROWHEAD_AUTHENTICATED_SYSTEM).toString();
+				
+				// if requester is blacklist, no need for check
+				if (!systemName.equals(Constants.SYS_NAME_BLACKLIST)) {
+						
+					final boolean isBlacklisted = arrowheadHttpService.consumeService(Constants.SERVICE_DEF_BLACKLIST_DISCOVERY, Constants.SERVICE_OP_CHECK, boolean.class, List.of(systemName));
+			
+					if (isBlacklisted) {
+						throw new ForbiddenException(systemName + " system is blacklisted!");
+					}
+				}
+			} catch (final ForbiddenException | AuthException ex) {
+				throw ex;
+			} catch (final ArrowheadException ex) {
+				logger.info("Blacklist server is not available.");
+				logger.debug("Blacklist server is not available, force blacklist filter: " + force);
+				if (force) {
+					throw new ForbiddenException("Blacklist system is not available, the system might be blacklisted.");
+				} else {
+					chain.doFilter(request, response);
+				}
```

Comment:
I think this is not necessary. The doFilter in the end will be called. Actually, in some cases your code calls doFilter twice, which is not good.